app/main.py

```diff
@@ -0,0 +1,474 @@
+
+            return device
+
+    def delete(self, key):
+        """Delete a key from Redis client."""
+        try:
+            response = self.redis.delete(key)
+            logging.info("response: %s", str(response))
+
+        except ConnectionError:
+            logging.exception("Connection error")
+        except Exception:
+            logging.exception("Unexpected error occured")
+        return response
+
+    def keys(self, param):
+        """Return list of all keys matching parameter."""
+        keys = []
+        try:
+            for key in self.redis.scan_iter(match=param+"*"):
+                logging.debug("key: %s", key)
+                keys.append(key.decode())
+
+        except ConnectionError:
+            logging.exception("Connection error")
+        except Exception:
+            logging.exception("Unexpected error occured")
+        return keys
+
+    def set_hb(self, heartbeat):
+        """Set heartbeat key in redis that expires in provided time."""
+        try:
+            response = self.redis.setex(
+                heartbeat, HB_EXP, "none")  # name, time, value
+            logging.info("heartbeat set")
+
+        except ConnectionError:
+            logging.exception("Connection error")
+        except Exception:
+            logging.exception("Unexpected error")
+        return response
+
+
+# Initialize REDIS object
+REDIS = Redis()
+
+
+#########################
+#   Heartbeat Methods   #
+#########################
+# @API.doc(params={'Authorization': {'in': 'header', 'description': 'Bearer ${api key}'}})
+@API.doc(responses={401: 'Unauthorized', 404: 'Incorrect request', 500: 'Server error'})
+@API.route('/devices/hb/<string:device_id>', methods=['GET', 'POST'])
+class Heartbeat(Resource):
+    """Update and check on a given device's heartbeat/online status."""
+
+    @validate_access
+    @API.response(200, 'Success')
+    def get(self, device_id):
+        """Check if a given heartbeat exists."""
+        logging.debug("checking device HB")
+        heartbeat = "hb_" + device_id
+        response = REDIS.keys(heartbeat)
+        logging.debug("HB get response = %s", str(response))
+        if response == []:
+            logging.error('Error, unable to get device.')
+            return ('Error, unable to get device', 500)
+
+        return jsonify(response)
+
+    @API.response(201, 'Success')
+    @API.expect(HEARTBEAT, validate=False)
+    @validate_access
+    def post(self, device_id):
+        """Set a heartbeat."""
+        heartbeat = "hb_" + device_id
+        response = REDIS.set_hb(heartbeat)
+        logging.info("HB post response = %s", response)
+        payload = API.payload
+        logging.info("payload = %s", payload)
+        if response:
+            logging.info("heartbeat set: %s", response)
+            return jsonify(response)
+
+        logging.error("Error, failed to set heartbeat")
+        return "Error, failed to set heartbeat.", 500
```

Comment:
In some places you seem to return the tuple `('Error, unable to get device', 500)` and in some you just return two values `"Error, failed to set heartbeat.", 500`.  

Even assuming they both work, I think it would be nice to standardize on one single convention.

Reply:
See master/new version.